### PR TITLE
fix: render twice and path resolution for production

### LINKED ISSUE #31: Lifecycle Hooks Are Called Twice And Bundle Paths Are Wrong In Production
## Describe the bug
Lifecycle hooks are called twice per route load. This makes logic in `mounted` and other lifecycle methods run twice per page load.

If using split bundles (as the example does,) some chunk paths point relative to `/_nuxt/`. This causes "chunk failed to load" errors when running the Nuxt application inside a host application.

## Expected behavior
Lifecycle hooks should only be called once per page load and bundles should resolve to their proper paths. For example, if my Nuxt single SPA is on localhost:80 but the root application is on localhost:5000, bundles should always load from localhost:80/_nuxt and not resolve to localhost:5000/_nuxt

### PR DESCRIPTION
**Pull request by munawarb**
Fixes #31

### CODE REVIEW

**Comment by munawarb** [comment]
Looks like if this approach is taken, then components don't receive props passed in from the root project, so I'm closing this as an ineffective solution.

File: examples/single-spa-demo/nuxt-subapp/nuxt.config.js [hunk 1]
```diff
@@ -72,6 +79,10 @@ export default {
   ** See https://nuxtjs.org/api/configuration-build/
   */
   build: {
+    // Make sure all generated bundles and chunks point to your server.
+    // If running in production meaning that the application will be built using npm run build && npm run start, point host to your domain.
+    // This is done in package.json.
+    publicPath: `${process.env.PROTOCOL}://${process.env.HOST}:${process.env.PORT}/_nuxt/`,
```
Comment (cjfff) [inline examples/single-spa-demo/nuxt-subapp/nuxt.config.js:85]: 这里网络协议为什么指定？构建时这个写死到 html 中

Comment (levy9527) [reply examples/single-spa-demo/nuxt-subapp/nuxt.config.js:85]: hi there, you can remove `${process.env.PROTOCOL}`，publicPath can start with `//` like: //google.com

File: examples/single-spa-demo/nuxt-subapp/mfe.js [hunk 2]
```diff
@@ -9,16 +15,13 @@ export default function(render) {
...
 }
 
 export async function update() {}
 
 export function mounted(instance, props) {
-  // inject your custom props
-  if (props.sdk) {
-    Vue.prototype.$sdk = props.sdk
-  }
+  addProps(props);
```
Comment (cjfff) [inline examples/single-spa-demo/nuxt-subapp/mfe.js:24]: why the addProps fucntion used in mounted and mount

Comment (munawarb) [reply examples/single-spa-demo/nuxt-subapp/mfe.js:24]: Hi @cjfff,
Without the `addProps` function in `mount`, props defined on the Vue instance will return undefined in child components. For example, `this.$sdk` would be undefined in a child component.

Previously, `mount` had a call to `render` in it, but this was causing double-calling of lifecycle hooks.

Comment (cjfff) [reply examples/single-spa-demo/nuxt-subapp/mfe.js:24]: The `$sdk` is undefined maybe cause by the vue lifecycle issue, you can read this [article](https://programmersought.com/article/99631733911/) to understand it.

**Review by cjfff — APPROVED** [review]

**Review by levy9527 — APPROVED** [review]

**Comment by levy9527** [comment]
@all-contributors add @munawarb bug

**Pull request merged**